src/backend/core/models.py

```diff
@@ -774,11 +951,27 @@ class Meta:
     def __str__(self):
         return self.title
 
+    def get_roles(self, user):
+        """Return the roles a user has on a resource."""
+        if not user.is_authenticated:
+            return []
+
+        try:
+            roles = self.user_roles or []
```

Comment:
Is the `or []` necessary? because if the attribute is present it should be `[]` already, no?

Reply:
it was set to None and we want to be able to use it with "in" statements.